docs: merge error related pages (#2198)
FuelLabs/fuels-ts: apps/docs/src/guide/errors/index.md

# Errors

> All errors thrown from our SDK are instances of the `FuelError` class, which will have an accompanying `ErrorCode`.

Reviewer: ```suggestion
All errors thrown from the SDK are instances of the `FuelError` class which will have an accompanying `ErrorCode`.
```

## Error Codes

Here is a list of the expected error codes the SDK can throw. These error codes are used to help understand the error that has been thrown with potential resolutions.

### `ABI_MAIN_METHOD_MISSING`

> When your ABI does not have a main method.

Reviewer: ```suggestion
When your ABI does not have a `main` method.
```

> This can be resolved by adding a main method to your ABI.

Reviewer: ```suggestion
This can be resolved by adding a `main` method to your ABI. This is prevalent in scripts and predicates that must contain a `main` method.
```

### `ABI_TYPES_AND_VALUES_MISMATCH`

When the arguments supplied to the function do not match the minimum required input length.

> Check that the arguments supplied to the function, match the required type.

Reviewer: ```suggestion
Check that the arguments supplied to the function match the required type.
```

### `ACCOUNT_REQUIRED`

When an [`Account`](../../api/Account/Account.md) is required for an operation. This will usually be in the form of a [`Wallet`](../wallets/index.md).

It could be caused during the deployments of contracts when an account is required to sign the transaction. This can be resolved by following the deployment guide [here](../contracts/deploying-contracts.md).

### `CONVERTING_FAILED`

When converting a big number into an incompatible format.

Ensure that the value you've supplied to the big number is compatible with the value you are converting to.

### `DUPLICATED_POLICY`

When there are more than policies with the same type, for a transaction.

Ensure that there are no duplicate (by type) policies for a transaction.

### `ERROR_BUILDING_BLOCK_EXPLORER_URL`

When more than one of the following options is passed: `path`, `address`, `txId`, `blockNumber`.

> Check that only one of the following options is passed: `path`, `address`, `txId`, `blockNumber`.

Reviewer: ```suggestion
Check that only one of the above is passed.
```

### `GAS_PRICE_TOO_LOW`

When the gas price is lower than the minimum gas price.

Increase the gas price to be greater than the minimum gas price.

### `HD_WALLET_ERROR`

A hardware wallet will throw for unsupported configurations.

The error message will determine which element of the configuration is incorrect. It could be due to the public or private key or when configuring to/from an extended key.

### `INVALID_CHECKSUM`

Checksum validation failed for the provided mnemonic.

Ensure that the mnemonic is correct.

### `INVALID_CONFIGURABLE_CONSTANTS`

When the program type either: does _not_ have configurable constants to be set; or the provided configurable constant does not belong to the program type, as defined by its ABI.

Ensure the configurable constants provided are correct and are defined in ABI.

### `INVALID_COMPONENT`

When an expected component is not found in the ABI or is malformed.

Ensure that you have correctly formed Sway types for [Arrays](../types/arrays.md) and [Vectors](../types/vectors.md).

### `INVALID_CREDENTIALS`

When the password provided is incorrect.

Ensure that the password is correct.

### `INVALID_DATA`

> When the value is more than the max value.
> 
> Ensure that the value is less than the max value.

Reviewer: ```suggestion
When the value being passed is not considered valid, as defined by the function.

Check the function signature and ensure that the passed value is valid.
```

### `INVALID_ENTROPY`

When the entropy is not: between 16 and 32 bytes; a multiple of 4.

Ensure that the entropy is between 16 and 32 bytes and a multiple of 4.

### `INVALID_EVM_ADDRESS`

When the provided EVM address is invalid.

Ensure that the [EVM address](../types/evm-address.md) is valid.

### `INVALID_INPUT_PARAMETERS`

When the provided input parameters are _not_ valid.

The error message will determine which parameter is missing. It could be that the provided program type is not one of the following `contract`, `script`, or `predicate`.

### `INVALID_MNEMONIC`

When the supplied mnemonic is invalid.

> Check the message for more details. It could be that the mnemonic phrase word length is \_not\_\_ one of the following: 12, 15, 18, 21, or 24 lengths.

Reviewer: ```suggestion
Check the message for more details. It could be that the mnemonic phrase word length is _not_ one of the following: 12, 15, 18, 21, or 24 lengths.
```

> 
> ### `INVALID_MULTICALL`
> 
> When the multi-call has a call that returns a heap type that is not the last call, or the multi-call has more than one call that returns a heap type.
> 
> Ensure that the multi-call has only one call that returns a heap type (which should be the last call).

Reviewer: ```suggestion
```

Not valid as of `v1` encoding.
